### DISCUSSION TITLE
Can tokio set the affinity of worker queues for task and disable the scheduling of steals?

### DISCUSSION THREAD

**Discussion by curtainp** [tokio-rs/tokio#5252] [Q&A]
Hi,
  I've come across a scenario where I need to distribute tasks based on certain properties of the task to ensure that tasks with the same properties are distributed to the same worker queue and are not stolen by other workers.

  The reason I needed this scenario was that I was creating tasks with the same properties that would cause lock competition, and I wanted to use the “current_thread” model to eliminate it and also get the advantage of multi_thread scheduling.

  I have consulted the tokio tutorial and recently traced it against the source code, and found that there is no relevant API interface provided.
  Can provide some assistance for me?

**Comment by Darksonn** [comment]
If you create multiple runtimes, then no work stealing happens between different runtimes. Additionally, CPU affinity can be set using [`Builder::on_thread_start`][1].

[1]: https://docs.rs/tokio/latest/tokio/runtime/struct.Builder.html#method.on_thread_start

**Reply by Noah-Kennedy** [reply]
This is the closest that we get, but I actually wish we had the ability to do something like what is being asked for. Inter-runtime communication is unfortunately not exactly efficient due to the need to pipe everything through the injector queue.

It would be nice to be able to have pinned tasks on that behave like they are on a separate runtime with respect to cross-worker wakes to them and can spawn tasks themselves that can be stolen, or wake other tasks which aren't pinned by pushing to the local queue as normally happens.

**Reply by curtainp** [reply]
I think this is exactly what I need, just for some special task, pin it to the local queue, the rest remains the same, in addition, I found a `LocalSet` structure, can this somehow approach this purpose?

**Reply by Noah-Kennedy** [reply]
Not really. It only works from the main/root task because that's a special case and actually doesn't run from a normal worker.
